src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/AutonomousManager.java

```diff
@@ -120,4 +128,79 @@ public SortedMap<String, String> getAllScanSettingsProperties() {
 
         return scanSettingsProperties;
     }
+
+    public void updateScanSettingsProperties(SortedMap<String, String> propertiesMap, List<String> adoptedScanTypes) {
+        userProvidedProperties.forEach((propertyKey, propertyValue) -> {
+            Optional<String> scanTypeValue = findScanType(adoptedScanTypes, propertyKey);
+            determinePropertyTypeAndUpdate(propertyKey, propertyValue, scanTypeValue, true);
+        });
+        propertiesMap.forEach((propertyKey, propertyValue) -> {
+            Optional<String> scanTypeValue = findScanType(adoptedScanTypes, propertyKey);
+            determinePropertyTypeAndUpdate(propertyKey, propertyValue, scanTypeValue, false);
+        });
+
+        scanSettings.setDetectorSharedProperties(detectorSharedProperties);
+        scanSettings.setGlobalDetectProperties(globalProperties);
+    }
+
+    private void determinePropertyTypeAndUpdate(String propertyKey, String propertyValue, Optional<String> scanTypeValue, boolean userProvidedProperty) {
+        if(isDetectorTypeProperty(propertyKey, propertyValue)) {
+            updateDetectorProperties(propertyKey, propertyValue, userProvidedProperty);
+        } else if (scanTypeValue.isPresent()) {
+            String scanTypeName = scanTypeValue.get();
+            ScanType scanType = scanSettings.getScanTypeWithName(scanTypeName);
+            if(userProvidedProperty && !propertyValue.isEmpty()) {
+                scanType.getScanProperties().put(propertyKey, propertyValue);
+            } else if(!scanType.getScanProperties().containsKey(propertyKey) && !propertyValue.isEmpty()) {
+                scanType.getScanProperties().put(propertyKey, propertyValue);
+            }
+        } else if(isGlobalTypeProperty(propertyKey, propertyValue)) {
+            updateGlobalProperties(propertyKey, propertyValue, userProvidedProperty);
+        }
+    }
```

Comment:
This function is called by the above method call to determine the type of bracket the property falls in. A user provided property means we update it anyhow, otherwise we update only if it is not present (To avoid case where default does not replace previously applied property): 

1. The first type is detectorSharedProperties which is selected if property contains detector in its key and DETECTOR tool is ran.
2. Second one is specific scanType property. For example, if NUGET scan was ran then all nuget properties, will be put under that bracket, similar for binary scan.
3. The third and last are the global properties which applies for the whole scan and are not sensitive.